File: Python/740. Delete and Earn.py
class Solution:
    def deleteAndEarn(self, nums: List[int]) -> int:
        n = 10001
        sum = [0]*n
        arr = [0]*n
        for i in nums:
            sum[i] += i
        
        arr[0] = 0
        arr[1] = sum[1]
        for i in range(2,n):
            arr[i] = max(arr[i-2] + sum[i], arr[i-1])
    
        return arr[n-1]

File: C#/740. Delete and Earn.cs
public class Solution {
    public int DeleteAndEarn(int[] nums) {
        int n = 10001;
        int[] sum = new int[n]; 
        int[] arr = new int[n]; 
        foreach(int i in nums){
            sum[i] += i;
        }
        arr[0] = 0; arr[1] = sum[1];
        for(int i=2;i<n;i++){
            arr[i] = Math.Max(arr[i-2] + sum[i], arr[i-1]);
        }
        return arr[n-1];
    }
}
